include/soci/soci-platform.h

```diff
@@ -18,13 +25,63 @@
 #ifdef _MSC_VER
 #include <stdlib.h>
 
+//Disables warnings about STL objects need to have dll-interface and/or
+//base class must have dll interface
+#pragma warning(disable:4251 4275)
+
 // Define if you have the vsnprintf variants.
 #if _MSC_VER < 1500
 # define vsnprintf _vsnprintf
 #endif
 
+#if _MSC_VER >= 1400 //use secure version removing compiler warnings
+    #define sprintf sprintf_s //just use secure version( same prototype )
+    #define sscanf sscanf_s //just use secure version( same prototype )
+    #define strncpy strncpy_secure //use inline wrapper bellow
+    #define strcpy strcpy_secure //use inline wrapper bellow
+    #define snprintf snprintf_secure //use inline wrapper bellow
+
+    inline char* strncpy_secure(char* buf, const char* src, size_t count)
+    {
+        errno_t err = strncpy_s(buf,count,src,count);
+        if( err != 0 )
+            return 0;
+        return buf;
+    }
+
+    inline char* strcpy_secure(char* dest, const char* src)
+    {
+        errno_t err = strcpy_s(dest,strlen(src)+1,src);
```

Comment:
On 8 July 2015 at 12:06, ravselj <notifications@github.com> wrote:oskot
<https://github.com/mloskot>, @vadz <https://github.com/vadz>
>
> 2. define _CRT_SECURE_NO_WARNINGS globally in soci-platform and then
> include soci-platform.h first
>
> I think second one is better since it results in cleaner code. What do you
> think?
>

IMO, yes, the 2. is a good idea as well as convention to include
soci-platform.h first.

Please, wait for @vadz final opinion as he may see issues I can't - I'm not
as concerned about
breaking users code as Vadim, but I respect that, so Vadim's got final word
here.